Add wordpress.com/stats/site deep link

zwarm opened the pull request:
Fixes #11967 

**Description**
This PR adds support for deep linking to the main (insights) stats tab from an external source/email. 

New `data` nodes were added to the manifest to handle https\http /stats paths. `DeepLinkingIntentReceiverActivity` was enhanced to handle the new stats deep link. A few helpers methods were added, as they are shared between existing handlers and stats, plus they can be reused as we add further deep links.

As with other deep links, if the user is not logged in when tapping on a stats link, they will be redirected to login. If a user is logged in, but the site doesn’t exist, the user will be taken to the stats tab for the current site. 

**To test:**

**Option A**
Using ADB
- Log into the app 
- Push the app to the background or close it
- Execute from a terminal
     ./adb shell am start -a android.intent.action.VIEW -c android.intent.category.BROWSABLE -d "https://wordpress.com/stats/annmarietestdomain.blog"

**Option B**
- Log into the app 
- Push the app to the background or close it
- Send yourself an email with a link to a site you have access to:https://wordpress.com/stats/{site url}
- Tap the link
- You may be prompted with a "open wordpress.com links with WordPress" dialog. If so, select an option. 
- The app should launch and you will be taken to the Insights stats tab
 
**Option C**
- Log into the app 
- Push the app to the background or close it
- Send yourself an email with a link to a site that is invalid: https://wordpress.com/stats/{fake site url}
- Tap the link
- You may be prompted with a "open wordpress.com links with WordPress" dialog. If so, select an option. 
- The app should take you to the current site Stats Insights tab

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

malinajirka commented:
Thanks @zwarm, great job! The code looks great and the app seems to work. I've encountered one bug, but it's very weird and I'm not sure if it was introduced in this PR.

![empty-screen](https://user-images.githubusercontent.com/2261188/82646193-70903780-9c14-11ea-8b1f-3fc10f676ef0.gif)
Steps
1. Close the app
2. Send yourself an email with a link to a site you have access to:https://wordpress.com/stats/{site url}
1. Tap the link
2. You may be prompted with a "open wordpress.com links with WordPress" dialog. If so, select an option.
3. The app should launch and you will be taken to the Insights stats tab
4. Press back twice (will close the app)
5. Tap the link in the email again
6. Notice an empty screen is shown

zwarm commented:
> I've encountered one bug, but it's very weird and I'm not sure if it was introduced in this PR.

@malinajirka Very odd indeed and I can not reproduce using this an emulator or a physical device. What specs are you running? Thanks.

malinajirka commented:
I was running stock emulator, API 27, Pixel 2.

I tried to run the same test on Nexus 9, API 29 and the result was a bit different. I think the issue might be related to the intent flags `Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK`.

![deeplink2](https://user-images.githubusercontent.com/2261188/82691833-de615100-9c5e-11ea-9943-392986ba5252.gif)

zwarm commented:
@malinajirka 
> I was running stock emulator, API 27, Pixel 2.

I was able to see the blank view on API 27, Pixel 2 stock emulator. A "finish" appears to clear that up in d2b5b84 if you want to take it for another whirl.

I also changed the merge branch into a feature-branch and removed the Draft label. I'll be adding another PR for sites pages.
